prometheus/testutil/testutil.go

```diff
@@ -221,6 +221,33 @@ func TransactionalGatherAndCompare(g prometheus.TransactionalGatherer, expected
 	return compareMetricFamilies(got, wanted, metricNames...)
 }
 
+// CollectAndFormat registers the provided Collector with a newly created pedantic Registry. It then calls Gather with
+// that Registry and filters by the provided metricNames. It writes the gathered metrics to a bugger formatted according
+// to the provided format argument
```

Comment:
Makes sense. I will be controversial here and suggest we stop doing so verbose comments. This comment explains implementation. I know you copied that for consistency, but we have to start somewhere. What about focusing on contract and usability. Implementation is trivial and self-described by the code. E.g.:

```suggestion
// CollectAndFormat collects the given `metricNames` metrics using pedantic registry and returns in the given format.
```

Do you mind fixing `GatherAndCompare` in the same manner, if that's ok with you?

Reply:
Yep, I'm OK with that